comments/models.py

```diff
@@ -1,23 +1,53 @@
 from django.db import models
+from django.utils import timezone
 from django_comments_xtd.models import XtdComment
 from wagtail.admin.edit_handlers import FieldPanel
 from wagtail.core.models import Page
 
+class CommentStatus:
+    OPEN = 'open'
+    CLOSED = 'closed'
+    DISABLED = 'disabled'
+    TIMESTAMPED = 'timestamped'
+
+    Choices = (
+        (OPEN, 'Open'),
+        (CLOSED, 'Closed'),
+        (DISABLED, 'Disabled'),
+        (TIMESTAMPED, 'Timestamped')
+    )
+
 
 class CommentableMixin(models.Model):
     """
     Add this mixin to any Model which wants to allow/disallow comments.
     Make sure you update get_absolute_url if this it hasn't already been
     included.
 
-    Use comments_panels to modify allow_comments from the admin.
+    Use comments_panels to modify commenting_status from the admin.
     """
-    allow_comments = models.BooleanField(default=False)
+    commenting_status = models.CharField(max_length=15, choices=CommentStatus.Choices)
+    commenting_starts_at = models.DateTimeField(null=True, blank=True)
+    commenting_ends_at = models.DateTimeField(null=True, blank=True)
+
 
     comments_panels = [
-        FieldPanel('allow_comments', heading='Enable?')
+        FieldPanel('commenting_status', heading='Status'),
+        FieldPanel('commenting_starts_at', heading='Commenting Opens At'),
+        FieldPanel('commenting_ends_at', heading='Commenting Ends At')
+
     ]
 
+    def should_show_comments_list(self):
+        return self.commenting_status in [CommentStatus.OPEN, CommentStatus.CLOSED, CommentStatus.TIMESTAMPED]
+
+    def should_show_new_comment_box(self):
+        commenting_still_valid = self.commenting_starts_at > timezone.now() > self.commenting_ends_at
```

Comment:
This throws an error if one of the time fields is left blank.
I'd suggest something like this (still needs fix to do timezone-aware/naive comparison):

```
        commenting_starts_at = self.commenting_starts_at or datetime.datetime.min
        commenting_ends_at = self.commenting_ends_at or datetime.datetime.max
        commenting_still_valid = commenting_starts_at < timezone.now() < commenting_ends_at
```

This way, e.g. leaving the commenting_ends field blank means commenting stays open indefinitely.